src/Workspaces/VisualBasic/Portable/FindSymbols/VisualBasicDeclaredSymbolInfoFactoryService.vb

```diff
@@ -291,6 +290,31 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.FindSymbols
             Return False
         End Function
 
+        Private Function IsExtensionMethod(node As MethodBlockSyntax) As Boolean
+            Dim parameterCount = node.SubOrFunctionStatement.ParameterList?.Parameters.Count
+
+            ' Extension method must have at least one parameter and declared inside a module
+            If Not parameterCount.HasValue Or parameterCount.Value = 0 Or TypeOf node.Parent IsNot ModuleBlockSyntax Then
+                Return False
+            End If
+
+            For Each attributeList In node.BlockStatement.AttributeLists
+                For Each attribute In attributeList.Attributes
+                    ' ExtensionAttribute takes no argument.
+                    If attribute.ArgumentList?.Arguments.Count > 0 Then
+                        Continue For
+                    End If
+
+                    Dim name = attribute.Name.ToString()
+                    If name.Equals("Extension") Or name.Equals("ExtensionAttribute") Or name.EndsWith(".Extension") Or name.EndsWith(".ExtensionAttribute") Then
```

Comment:
don't ToString the name.  based on code i think you've already written, you can get the last name portion of attribute.name.  so just call that helper and check that.

Reply:
fixed